syntax = "proto3";

package SpravSerialize;

message Bus {
  uint32 id = 1;
  string name = 2;

  uint32 stops_count = 5;
  uint32 unique_stops_count = 6;
  uint32 length = 7;
  float curvature = 8;
}

message Stop {
  uint32 id = 1;
  string name = 4;

  repeated uint32 bus = 5;
}

message Edge {
  uint32 id = 1;
  uint32 from = 2;
  uint32 to = 3;
  float weight = 4;

  repeated uint32 incidence_list = 5;
}

message Graph {
  repeated Edge edge = 1;
}

message Router {
}

message TransportCatalog {
  repeated Bus bus = 1;
  repeated Stop stop = 2;
  Graph graph = 3;
  Router router = 4;
}